### DISCUSSION TITLE
Possible to have 'choice' type arguments?

### DISCUSSION THREAD

**Discussion by gCardinal** [jmcdo29/nest-commander#217] [Q&A] [answered]
I'd like to have an argument for a command only allow a list of valid values. [Exactly like this commander example](https://github.com/tj/commander.js/blob/HEAD/examples/arguments-extra.js). Is this something possible with nest-commander?

**Comment by jmcdo29** [comment, marked as answer]
You could simulate this with a [`required` option](https://jmcdo29.github.io/nest-commander/docs/api/#option) which will pretty much act like an argument, but come in through the `options` parameter instead. 

I can look into how to add this functionality otherwise, but it will probably be a breaking change to make the current commander API line up better with nest-commander's API

**Reply by gCardinal** [reply]
I opted for a custom validator service on the string I receive, but your method would have also worked. That being said, it would be great to have the ability for more robust arguments in nest-commander!
